types/net_fields.d.ts

```diff
@@ -1,323 +1,37 @@
-  x: 6;
-  y: 7;
-  z: 8;
-};
-
-// 0x1D, copied from emulator
-type NetCombatantWaymarkerFields = Fields & {
-  operation: 2;
-  waymark: 3;
-  id: 4;
-  name: 5;
-  targetId: 6;
-  targetName: 7;
-};
-
-// 0x1E
-type NetLosesEffectFields = Fields & {
-  effectId: 2;
-  effect: 3;
-  sourceId: 5;
-  source: 6;
-  targetId: 7;
-  target: 8;
-  count: 9;
-};
-
-// 0x1F, copied from emulator
-type NetJobGaugeFields = Fields & {
-  id: 2;
-  dataBytes1: 3;
-  dataBytes2: 4;
-  dataBytes3: 5;
-  dataBytes4: 6;
-};
-
-// 0x21
-type NetActorControlFields = Fields & {
-  instance: 2;
-  command: 3;
-  data0: 4;
-  data1: 5;
-  data2: 6;
-  data3: 7;
-};
-
-// 0x22
-type NetNameToggleFields = Fields & {
-  id: 2;
-  name: 3;
-  targetId: 4;
-  targetName: 5;
-  toggle: 6;
-};
-
-// 0x23
-type NetTetherFields = Fields & {
-  sourceId: 2;
-  source: 3;
-  targetId: 4;
-  target: 5;
-  id: 8;
-};
-
-// 0x24, copied from emulator
-type NetLimitGaugeFields = Fields & {
-  valueHex: 2;
-  bars: 3;
-};
-
-// 0x25, copied from emulator
-type NetActionSyncFields = Fields & {
-  id: 2;
-  name: 3;
-  sequenceId: 4;
-  currentHp: 5;
-  maxHp: 6;
-  currentMp: 7;
-  maxMp: 8;
-  // currentTp: 9;
-  // maxTp: 10;
-  x: 11;
-  y: 12;
-  z: 13;
-  heading: 14;
-};
-
-// 0x26
-type NetStatusEffectFields = Fields & {
-  targetId: 2;
-  target: 3;
-  hp: 5;
-  maxHp: 6;
-  x: 11;
-  y: 12;
-  z: 13;
-  heading: 14;
-  data0: 15;
-  data1: 16;
-  data2: 17;
-  data3: 18;
-  data4: 19;
-};
-
-// 0x40
-type NetMapFields = Fields & {
-  id: 2;
-  regionName: 3;
-  placeName: 4;
-  placeNameSub: 5;
-};
-
+// TODO: add a netFieldsName to netlog_defs so this can be generated generically?
 export type NetFields = {
-  'GameLog': NetGameLogFields;
-  'ChangeZone': NetChangeZoneFields;
-  'ChangedPlayer': NetChangedPlayerFields;
-  'AddedCombatant': NetAddedCombatantFields;
-  'RemovedCombatant': NetRemovedCombatantFields;
-  'PlayerStats': NetPlayerStatsFields;
-  'StartsUsing': NetStartsUsingFields;
-  'Ability': NetAbilityFields;
-  'AOEAbility': NetAOEAbilityFields;
-  'CancelAbility': NetCancelAbilityFields;
-  'DoTHoT': NetDoTHoTFields;
-  'WasDefeated': NetWasDefeatedFields;
-  'GainsEffect': NetGainsEffectFields;
-  'HeadMarker': NetHeadMarkerFields;
-  'FloorWaymarker': NetFloorWaymarkerFields;
-  'CombatantWaymarker': NetCombatantWaymarkerFields;
-  'LosesEffect': NetLosesEffectFields;
-  'JobGauge': NetJobGaugeFields;
-  'ActorControl': NetActorControlFields;
-  'NameToggle': NetNameToggleFields;
-  'Tether': NetTetherFields;
-  'LimitGauge': NetLimitGaugeFields;
-  'ActionSync': NetActionSyncFields;
-  'StatusEffect': NetStatusEffectFields;
-  'Map': NetMapFields;
+  'GameLog': typeof logDefinitions.gameLog.fields;
+  'ChangeZone': typeof logDefinitions.changeZone.fields;
+  'ChangedPlayer': typeof logDefinitions.changePrimaryPlayer.fields;
+  'AddedCombatant': typeof logDefinitions.addCombatant.fields;
+  'RemovedCombatant': typeof logDefinitions.addCombatant.fields;
```

Comment:
If you elect not to implement my suggestion, this looks like a bug (both are `addCombatant`)?

Reply:
I'll change it.  The reasoning was because:
```
// 0x04, identical to 0x03 for fields
type NetRemovedCombatantFields = NetAddedCombatantFields;
```